File: problems/20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<>();
        java.util.Map<Character, Character> d = new java.util.HashMap<>();
        d.put(')', '(');
        d.put('}', '{');
        d.put(']', '[');
        for (char i:s.toCharArray()) {
            if (i=='(' || i=='{' || i=='[')
                stk.push(i);
            else if (i==')' || i=='}' || i==']'){
                if (!stk.isEmpty() && stk.peek()==d.get(i))
                    stk.pop();
                else
                    return false;
            }
        }
        return stk.isEmpty();
    }
}

File: problems/20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stk=[]
        d={')':'(','}':'{',']':'['}
        for i in s:
            if i=='(' or i=='{' or i=='[':
                stk.append(i)
            elif i==')' or i=='}' or i==']':
                if len(stk)>0 and stk[-1]==d[i]:
                    stk.pop()
                else:
                    return False
        return stk==[]